trio/_util.py

```diff
@@ -85,6 +88,87 @@ def is_main_thread():
         return False
 
 
+######
+# Call the function and get the coroutine object, while giving helpful
+# errors for common mistakes. Returns coroutine object.
+######
+def coroutine_or_error(async_fn, args=[]):
```

Comment:
Trio convention is to take `*args` in this case -- we almost never pass around the argument tuple as a tuple, even internally.